Random issue trying to deploy

I'm seeing a random issue while deploying. I didn't changed anything but sometimes the deployment pipeline on github is failing:

This is the log I'm seeing:

Run bun i
  bun i
  bunx sst deploy --stage beta
  shell: /usr/bin/bash -e {0}
  env:
    ....
bun install v1.3.5 (1e86cebd)

+ @tsconfig/node22@22.0.5
+ @types/aws-lambda@8.10.159
+ @types/bun@1.3.5
+ @types/jspdf@1.3.3
+ @types/koa@3.0.1
+ @types/koa__router@12.0.5
+ baseline-browser-mapping@2.9.11
+ typescript@5.9.3
+ @ai-sdk/amazon-bedrock@4.0.0
+ @ai-sdk/openai@3.0.0
+ @ai-sdk/react@3.0.1
+ @aws-sdk/client-bedrock-agent@3.955.0
+ @aws-sdk/client-bedrock-agent-runtime@3.955.0
+ @aws-sdk/client-cognito-identity-provider@3.955.0
+ @aws-sdk/client-dynamodb@3.955.0
+ @aws-sdk/client-lambda@3.955.0
+ @aws-sdk/client-s3@3.955.0
+ @aws-sdk/client-secrets-manager@3.956.0
+ @aws-sdk/client-ses@3.956.0
+ @aws-sdk/client-sqs@3.956.0
+ @aws-sdk/credential-providers@3.955.0
+ @aws-sdk/lib-dynamodb@3.955.0
+ @aws-sdk/s3-request-presigner@3.955.0
+ @deepgram/sdk@4.11.3
+ @formulajs/formulajs@4.5.5
+ @pdfme/common@5.5.8
+ @pdfme/manipulator@5.5.8
+ @slack/web-api@7.13.0
+ @tanstack/react-table@8.21.3
+ ai@6.0.1
+ axios@1.13.2
+ bun@1.3.5
+ csvtojson@2.0.14
+ dayjs@1.11.19
+ exa-js@1.10.2
+ googleapis@150.0.1
+ ical.js@2.2.1
+ iron-session@8.0.4
+ jsonata@2.1.0
+ jspdf@2.5.2
+ jspdf-autotable@5.0.2
+ lambda-stream@0.6.0
+ lucide-react@0.545.0
+ mathjs@15.1.0
+ mysql2@3.16.0
+ openai@6.8.1
+ pdf-lib@1.17.1
+ pdf-parse@2.4.5
+ pdfjs-dist@5.4.449
+ raw-loader@4.0.2
+ react-hook-form@7.68.0
+ react-icons@5.5.0
+ react-number-format@5.4.4
+ sharp@0.34.5
+ sonner@2.0.7
+ sst@3.17.25
+ truncate-json@3.0.1
+ twilio@5.11.1
+ xlsx@0.18.5
+ xlsx-calc@0.9.2
+ zod@3.25.76

1481 packages installed [20.47s]
SST 3.17.25  ready!

➜  App:        volto
   Stage:      beta

~  Deploy

flag provided but not defined: -root
Usage of /usr/local/bin/pulumi-language-nodejs:
  -alsologtostderr
    	log to standard error as well as files
  -log_backtrace_at value
    	when logging hits line file:N, emit a stack trace
  -log_dir string
    	If non-empty, write log files in this directory
  -log_link string
    	If non-empty, add symbolic links in this directory to the log files
  -logbuflevel int
    	Buffer log messages logged at this level or lower (-1 means don't buffer; 0 means buffer INFO only; ...). Has limited applicability on non-prod platforms.
  -logtostderr
    	log to standard error instead of files
  -stderrthreshold value
    	logs at or above this threshold go to stderr (default 2)
  -tracing string
    	Emit tracing to a Zipkin-compatible tracing endpoint
  -v value
    	log level for V logs
  -version
    	Print the current plugin version and exit
  -vmodule value
    	comma-separated list of pattern=N settings for file-filtered logging

Chiming in to say my team is facing the same issue.  Our production deployment is broken while we are trying to expedite a release.  Thanks all for the workarounds described above, will give it a try.

```
sst deploy --stage production

SST 3.17.8  ready!

➜  App:        smart-modules
   Stage:      production

~  Deploy

flag provided but not defined: -root
Usage of /usr/local/bin/pulumi-language-nodejs:
  -alsologtostderr
    	log to standard error as well as files
  -log_backtrace_at value
    	when logging hits line file:N, emit a stack trace
  -log_dir string
    	If non-empty, write log files in this directory
  -log_link string
    	If non-empty, add symbolic links in this directory to the log files
  -logbuflevel int
    	Buffer log messages logged at this level or lower (-1 means don't buffer; 0 means buffer INFO only; ...). Has limited applicability on non-prod platforms.
  -logtostderr
    	log to standard error instead of files
  -stderrthreshold value
    	logs at or above this threshold go to stderr (default 2)
  -tracing string
    	Emit tracing to a Zipkin-compatible tracing endpoint
  -v value
    	log level for V logs
  -version
    	Print the current plugin version and exit
  -vmodule value
    	comma-separated list of pattern=N settings for file-filtered logging

↗  Permalink   https://sst.dev/u/XXXXXXXXX

✕  Interrupted    

 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

--- a/pkg/project/run.go
+++ b/pkg/project/run.go
@@ -249,6 +249,13 @@ func (p *Project) RunNext(ctx context.Context, input *StackInput) error {
 	}
 
 	env := os.Environ()
+	// Prepend SST's bin directory to PATH to ensure SST's Pulumi is used over system-installed versions
+	for i, e := range env {
+		if strings.HasPrefix(e, "PATH=") {
+			env[i] = "PATH=" + global.BinPath() + string(os.PathListSeparator) + strings.TrimPrefix(e, "PATH=")
+			break
+		}
+	}
 	for key, value := range p.Env() {
 		env = append(env, fmt.Sprintf("%v=%v", key, value))
 	}

